llvm/test/Transforms/SCCP/pointer-nonnull.ll

```diff
@@ -210,18 +210,22 @@ define internal i1 @ip_test_nonnull_callee(ptr nonnull %p) {
 ;
 ; IPSCCP-LABEL: define internal i1 @ip_test_nonnull_callee(
 ; IPSCCP-SAME: ptr nonnull [[P:%.*]]) {
-; IPSCCP-NEXT:    [[CMP:%.*]] = icmp ne ptr [[P]], null
-; IPSCCP-NEXT:    ret i1 [[CMP]]
+; IPSCCP-NEXT:    ret i1 poison
```

Comment:
Is it correct? Shouldn't it be `true`?

Reply:
This is because IPSCCP replaces the the use of the return value in the caller, so the return value in the callee becomes unused and is replaced with poison.

Reply:
For IPSCCP, returned value isn't used any longer (at call sites we replaced by `true`)